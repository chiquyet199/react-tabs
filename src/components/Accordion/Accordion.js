import React, {Component} from "react";
import {TouchableOpacity, View} from "react-native";
import {bool, func} from "prop-types";
import Collapsible from "react-native-collapsible";
import {isWeb} from "../../utils/common";

export const CompatibleCollapsible = props => {
  return isWeb ? (
    <View>{props.collapsed ? null : props.children}</View>
  ) : (
    <Collapsible {...props}>{props.children}</Collapsible>
  );
};
CompatibleCollapsible.propTypes = {
  collapsed: bool,
};

export class Accordion extends Component {
  state = {
    collapsed: true,
  };

  handlePress = () => {
    this.setState({
      collapsed: !this.state.collapsed,
    });
  };

  render() {
    const {title: Title} = this.props;

    return (
      <TouchableOpacity onPress={this.handlePress}>
        <Title collapsed={this.state.collapsed} />
        <CompatibleCollapsible collapsed={this.state.collapsed}>
          {this.props.children}
        </CompatibleCollapsible>
      </TouchableOpacity>
    );
  }
}

Accordion.propTypes = {
  title: func.isRequired,
};

export default Accordion;
