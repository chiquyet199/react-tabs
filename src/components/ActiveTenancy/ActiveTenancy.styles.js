import styledNative from "styled-components/native";
import {themed} from "../../utils/common";
import Icon from "../Icon";

export const ThemedText = themed(styledNative.Text`
  color: ${({globaltheme, theme}) => theme[globaltheme].textColor};
`);
export const HorizontalStack = styledNative.View`
  flex-direction: row;
`;
export const SectionTitle = styledNative(ThemedText)`
  font-size: 24;
`;
export const SectionText = styledNative(ThemedText)`
  font-size: 18;
`;
export const SectionTextSmall = styledNative(ThemedText)`
  font-size: 16;
`;
export const Section = themed(styledNative.View`
  padding: 11px 0 13px;
  margin: 0 17px 0 18px;
  border-bottom-color: ${({globaltheme, theme}) =>
    theme[globaltheme].boxBorder};
  border-bottom-width: 1;
`);
export const ThemedIcon = themed(styledNative(Icon)`
  color: ${({globaltheme, theme}) => theme[globaltheme].textColor};
`);
