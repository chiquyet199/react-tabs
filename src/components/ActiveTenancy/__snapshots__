/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/ActiveTenancy/__native_snapshots__