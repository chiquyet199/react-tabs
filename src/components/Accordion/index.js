import Accordion from "./Accordion";
import AccordionWithCount from "./AccordionWithCount";
import AccordionWithIconTitle from "./AccordionWithIconTitle";

export {Accordion, AccordionWithCount, AccordionWithIconTitle};
