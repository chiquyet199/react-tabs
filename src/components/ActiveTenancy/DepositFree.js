import React, {Fragment} from "react";
import {connect} from "react-redux";
import {shape, string} from "prop-types";
import styledNative from "styled-components/native";
import LeaseSpec from "../../utils/api/specs/LeaseSpec";
import {Accordion, AccordionWithIconTitle} from "../Accordion";
import {
  HorizontalStack,
  Section,
  SectionTitle,
  SectionText,
  ThemedIcon,
} from "./ActiveTenancy.styles";
import {keyFromString} from "../../utils/string";

const DepositDocumentDetails = styledNative.View`
  flex-grow: 1;
`;

export const DepositFree = ({lease, translations}) => (
  <Accordion
    title={({collapsed}) => (
      <AccordionWithIconTitle
        headerIconName="claims"
        title={translations.title}
        subTitle={translations.subTitle}
        collapsed={collapsed}
      />
    )}
  >
    <Fragment>
      {lease.documents.map(document => (
        <Section key={keyFromString(document.type)}>
          <HorizontalStack>
            <DepositDocumentDetails>
              <SectionTitle>{translations.title}</SectionTitle>
              <SectionText>
                {translations.documentTypes[document.type]}
              </SectionText>
            </DepositDocumentDetails>
            <ThemedIcon name="legal" size={32} />
          </HorizontalStack>
        </Section>
      ))}
    </Fragment>
  </Accordion>
);

DepositFree.propTypes = {
  lease: LeaseSpec,
  translations: shape({
    subTitle: string.isRequired,
    title: string.isRequired,
    documentTypes: shape({
      POLICY_SCHEDULE: string.isRequired,
      STATEMENT_OF_FACTS: string.isRequired,
      POLICY_WORDING: string.isRequired,
      INSURANCE_PRODUCT_INFORMATION_DOCUMENT: string.isRequired,
      INSURANCE_SERVICE_INFORMATION_DOCUMENT: string.isRequired,
      AST: string.isRequired,
    }),
  }),
};

const mapStateToProps = state => ({
  translations: state.locale.translations.common.depositFree,
});

export default connect(mapStateToProps)(DepositFree);
