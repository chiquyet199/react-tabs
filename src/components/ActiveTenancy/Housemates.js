import React, {Fragment} from "react";
import {connect} from "react-redux";
import {shape, string} from "prop-types";
import styledNative from "styled-components/native";
import LeaseSpec from "../../utils/api/specs/LeaseSpec";
import {Accordion, AccordionWithIconTitle} from "../Accordion";
import {
  HorizontalStack,
  Section,
  ThemedIcon,
  ThemedText,
} from "./ActiveTenancy.styles";

const UserName = styledNative(ThemedText)`
  font-size: 16;
  font-weight: bold;
`;

export const Housemates = ({lease, translations}) => (
  <Accordion
    title={({collapsed}) => (
      <AccordionWithIconTitle
        headerIconName="group"
        title={translations.title}
        subTitle={translations.subTitle}
        collapsed={collapsed}
      />
    )}
  >
    <Fragment>
      {lease.renters.map(renter => (
        <Section key={renter.userId}>
          <HorizontalStack>
            <ThemedIcon name="rent-passport" size={46} />
            <UserName>{`${renter.firstName} ${renter.lastName}`}</UserName>
          </HorizontalStack>
        </Section>
      ))}
    </Fragment>
  </Accordion>
);

Housemates.propTypes = {
  lease: LeaseSpec,
  translations: shape({
    title: string.isRequired,
    subTitle: string.isRequired,
  }).isRequired,
};

const mapStateToProps = state => ({
  translations: state.locale.translations.common.housemates,
});

export default connect(mapStateToProps)(Housemates);
