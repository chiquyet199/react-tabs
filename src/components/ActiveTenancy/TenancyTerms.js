import React, {Fragment} from "react";
import {connect} from "react-redux";
import {shape, string} from "prop-types";
import LeaseSpec from "../../utils/api/specs/LeaseSpec";
import {Accordion, AccordionWithIconTitle} from "../Accordion";
import {
  Section,
  SectionTitle,
  SectionText,
  SectionTextSmall,
} from "./ActiveTenancy.styles";
import moment from "moment";

const getTenancyDates = (lease, ongoing) => {
  const {
    duration: {startDate, durationMonths},
  } = lease;
  const dateFormat = "DD.MM.YYYY";
  const formattedStartDate = moment(startDate).format(dateFormat);
  const endDate = durationMonths
    ? moment(startDate)
        .add(durationMonths, "M")
        .format(dateFormat)
    : ongoing;

  return `${formattedStartDate} - ${endDate}`;
};

export const TenancyTerms = ({lease, translations}) => (
  <Accordion
    title={({collapsed}) => (
      <AccordionWithIconTitle
        headerIconName="legal"
        title={translations.title}
        subTitle={translations.subTitle}
        collapsed={collapsed}
      />
    )}
  >
    <Fragment>
      <Section>
        <SectionTitle>{translations.totalRentTitle}</SectionTitle>
        <SectionText>
          {translations.currency_symbol +
            lease.rentAmount +
            translations[lease.rentFrequency]}
        </SectionText>
      </Section>
      <Section>
        <SectionTitle>{translations.termTitle}</SectionTitle>
        <SectionText>
          {getTenancyDates(lease, translations.ongoing)}
        </SectionText>
      </Section>
      <Section>
        <SectionTitle>{translations.additionalInfoTitle}</SectionTitle>
        <SectionTextSmall>{lease.conditionsFreeText}</SectionTextSmall>
      </Section>
    </Fragment>
  </Accordion>
);
TenancyTerms.propTypes = {
  lease: LeaseSpec,
  translations: shape({
    additionalInfoTitle: string.isRequired,
    currency_symbol: string.isRequired,
    ongoing: string.isRequired,
    subTitle: string.isRequired,
    termTitle: string.isRequired,
    title: string.isRequired,
    totalRentTitle: string.isRequired,
    tenancyPeriodAbbreviations: shape({
      WEEKLY: string.isRequired,
      FORTNIGHTLY: string.isRequired,
      FOUR_WEEKLY: string.isRequired,
      MONTHLY: string.isRequired,
      YEARLY: string.isRequired,
    }),
  }),
};

const mapStateToProps = state => ({
  translations: {
    currency_symbol: state.locale.translations.common.currency_symbol,
    ...state.locale.translations.common.tenancyTerms,
  },
});

export default connect(mapStateToProps)(TenancyTerms);
