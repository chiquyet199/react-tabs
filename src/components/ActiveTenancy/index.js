import DepositFree from "./DepositFree";
import Housemates from "./Housemates";
import TenancyTerms from "./TenancyTerms";

export {DepositFree, Housemates, TenancyTerms};
