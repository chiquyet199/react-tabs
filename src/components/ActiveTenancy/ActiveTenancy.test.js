import React from "react";
import {Platform} from "react-native";
import {shallow} from "enzyme";
import {TenancyTerms} from "./TenancyTerms";
import {Accordion} from "../Accordion";
import {SectionText} from "./ActiveTenancy.styles";
import {DepositFree} from "./DepositFree";
import {Housemates} from "./Housemates";

describe("Active tenancy component tests", () => {
  describe("TenancyTerms tests", () => {
    it(`matches snapshot on ${Platform.OS}`, () => {
      expect(getComponent()).toMatchSnapshot();
    });

    it("displays end date when it's provided", () => {
      const lease = getMockLease();
      lease.duration.startDate = "2018-12-12";
      lease.duration.durationMonths = 12;
      const children = getAccordionChildren({lease});
      expect(children.find(SectionText).at(1)).toHaveProp(
        "children",
        "12.12.2018 - 12.12.2019",
      );
    });

    it("displays ongoing when end date isn't provided", () => {
      const lease = getMockLease();
      const ongoing = "lorem ipsum";
      lease.duration.startDate = "2018-12-12";
      const children = getAccordionChildren({
        lease,
        translations: getTranslations({ongoing}),
      });
      expect(children.find(SectionText).at(1)).toHaveProp(
        "children",
        `12.12.2018 - ${ongoing}`,
      );
    });

    const getAccordionChildren = props => {
      const component = getComponent(props);

      return component.find(Accordion).dive();
    };

    const getComponent = props => {
      const parsedProps = {
        lease: getMockLease(),
        translations: getTranslations(),
        ...props,
      };

      return shallow(<TenancyTerms {...parsedProps} />);
    };

    const getTranslations = translations => ({
      additionalInfoTitle: "additionalInfoTitle",
      currency_symbol: "currency_symbol",
      ongoing: "ongoing",
      subTitle: "subTitle",
      termTitle: "termTitle",
      title: "title",
      totalRentTitle: "totalRentTitle",
      tenancyPeriodAbbreviations: {
        WEEKLY: "WEEKLY",
        FORTNIGHTLY: "FORTNIGHTLY",
        FOUR_WEEKLY: "FOUR_WEEKLY",
        MONTHLY: "MONTHLY",
        YEARLY: "YEARLY",
      },
      ...translations,
    });
  });

  describe("DepositFree tests", () => {
    it(`matches snapshot on ${Platform.OS}`, () => {
      expect(getComponent()).toMatchSnapshot();
    });

    const getComponent = props => {
      const parsedProps = {
        lease: getMockLease(),
        translations: getTranslations(),
        ...props,
      };

      return shallow(<DepositFree {...parsedProps} />);
    };

    const getTranslations = translations => ({
      subTitle: "depositFreeSubTitle",
      title: "depositFreeTitle",
      documentTypes: {
        POLICY_SCHEDULE: "POLICY_SCHEDULE",
        STATEMENT_OF_FACTS: "STATEMENT_OF_FACTS",
        POLICY_WORDING: "POLICY_WORDING",
        INSURANCE_PRODUCT_INFORMATION_DOCUMENT:
          "INSURANCE_PRODUCT_INFORMATION_DOCUMENT",
        INSURANCE_SERVICE_INFORMATION_DOCUMENT:
          "INSURANCE_SERVICE_INFORMATION_DOCUMENT",
        AST: "AST",
      },
      ...translations,
    });
  });

  describe("Housemates tests", () => {
    it(`matches snapshot on ${Platform.OS}`, () => {
      expect(getComponent()).toMatchSnapshot();
    });

    const getComponent = props => {
      const parsedProps = {
        lease: getMockLease(),
        translations: {
          title: "title",
          subTitle: "subtitle",
        },
        ...props,
      };

      return shallow(<Housemates {...parsedProps} />);
    };
  });
});

const getMockLease = () => ({
  id: "abc-123",
  branchId: "def-456",
  property: getMockProperty(),
  renters: [
    {
      userId: "ghi-789",
      depositOptions: [{type: "DEPOSIT_FREE"}],
      firstName: "Bill",
      lastName: "Oddy",
    },
    {
      userId: "ghi-987",
      depositOptions: [{type: "DEPOSIT_FREE"}],
      firstName: "Damon",
      lastName: "Albarn",
    },
  ],
  status: "ACTIVE",
  rentAmount: 1234,
  rentFrequency: "MONTHLY",
  duration: {
    minimumMonths: 6,
    startDate: "2018-11-06T00:00:00",
  },
  conditionsFreeText: "",
  documents: [
    {
      type: "POLICY_SCHEDULE",
      ref: "ghi-998",
      createdAt: "2018-06-01T15:32:48",
    },
    {
      type: "STATEMENT_OF_FACTS",
      ref: "ghi-712",
      createdAt: "2018-06-01T15:48:48",
    },
  ],
});

const getMockProperty = () => ({
  id: "abc-123",
  presentation: {
    bathroomsCount: 1,
    bedroomsCount: 2,
    furnishedType: "PART_FURNISHED",
    hasOutsideSpace: true,
    hasParking: true,
    petsAllowed: false,
    propertyType: "HOUSE",
    smokingAllowed: false,
  },
  rentFrequency: "WEEKLY",
  rentAmount: 1234.99,
  depositAmount: 1000,
  depositFreeAvailable: true,
  propertyImages: [
    {
      id: "def-456",
      url: "http://www.findyourcanopy.com/img123.jpeg",
      isPrimary: false,
    },
    {
      id: "def-457",
      url:
        "https://media.rightmove.co.uk/dir/152k/151109/61705537/151109_344038-1_IMG_19_0002_max_656x437.jpg",
      isPrimary: true,
    },
    {
      id: "def-458",
      url: "http://www.findyourcanopy.com/img125.jpeg",
      isPrimary: false,
    },
  ],
  landlord: {},
  address: {
    line1: "29 Acacia Road",
    town: "Beanotown",
    state: "Aberdeenshire",
    postCode: "BE1 1BE",
    countryCode: "UK",
  },
  createdBy: "Homogenous cohorts",
  agencyId: "22",
  branchId: "44",
});
