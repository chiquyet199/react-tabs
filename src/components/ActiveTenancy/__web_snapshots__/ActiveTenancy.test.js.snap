// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active tenancy component tests DepositFree tests matches snapshot on web 1`] = `
<Accordion
  title={[Function]}
>
  <Connect(WithTheme(Styled(View)))
    key="POLICY_SCHEDULE"
  >
    <Styled(View)>
      <Styled(View)>
        <Styled(Connect(WithTheme(Styled(Text))))>
          depositFreeTitle
        </Styled(Connect(WithTheme(Styled(Text))))>
        <Styled(Connect(WithTheme(Styled(Text))))>
          POLICY_SCHEDULE
        </Styled(Connect(WithTheme(Styled(Text))))>
      </Styled(View)>
      <Connect(WithTheme(Styled(Icon)))
        name="legal"
        size={32}
      />
    </Styled(View)>
  </Connect(WithTheme(Styled(View)))>
  <Connect(WithTheme(Styled(View)))
    key="STATEMENT_OF_FACTS"
  >
    <Styled(View)>
      <Styled(View)>
        <Styled(Connect(WithTheme(Styled(Text))))>
          depositFreeTitle
        </Styled(Connect(WithTheme(Styled(Text))))>
        <Styled(Connect(WithTheme(Styled(Text))))>
          STATEMENT_OF_FACTS
        </Styled(Connect(WithTheme(Styled(Text))))>
      </Styled(View)>
      <Connect(WithTheme(Styled(Icon)))
        name="legal"
        size={32}
      />
    </Styled(View)>
  </Connect(WithTheme(Styled(View)))>
</Accordion>
`;

exports[`Active tenancy component tests Housemates tests matches snapshot on web 1`] = `
<Accordion
  title={[Function]}
>
  <Connect(WithTheme(Styled(View)))
    key="ghi-789"
  >
    <Styled(View)>
      <Connect(WithTheme(Styled(Icon)))
        name="rent-passport"
        size={46}
      />
      <Styled(Connect(WithTheme(Styled(Text))))>
        Bill Oddy
      </Styled(Connect(WithTheme(Styled(Text))))>
    </Styled(View)>
  </Connect(WithTheme(Styled(View)))>
  <Connect(WithTheme(Styled(View)))
    key="ghi-987"
  >
    <Styled(View)>
      <Connect(WithTheme(Styled(Icon)))
        name="rent-passport"
        size={46}
      />
      <Styled(Connect(WithTheme(Styled(Text))))>
        Damon Albarn
      </Styled(Connect(WithTheme(Styled(Text))))>
    </Styled(View)>
  </Connect(WithTheme(Styled(View)))>
</Accordion>
`;

exports[`Active tenancy component tests TenancyTerms tests matches snapshot on web 1`] = `
<Accordion
  title={[Function]}
>
  <Connect(WithTheme(Styled(View)))>
    <Styled(Connect(WithTheme(Styled(Text))))>
      totalRentTitle
    </Styled(Connect(WithTheme(Styled(Text))))>
    <Styled(Connect(WithTheme(Styled(Text))))>
      currency_symbol1234undefined
    </Styled(Connect(WithTheme(Styled(Text))))>
  </Connect(WithTheme(Styled(View)))>
  <Connect(WithTheme(Styled(View)))>
    <Styled(Connect(WithTheme(Styled(Text))))>
      termTitle
    </Styled(Connect(WithTheme(Styled(Text))))>
    <Styled(Connect(WithTheme(Styled(Text))))>
      06.11.2018 - ongoing
    </Styled(Connect(WithTheme(Styled(Text))))>
  </Connect(WithTheme(Styled(View)))>
  <Connect(WithTheme(Styled(View)))>
    <Styled(Connect(WithTheme(Styled(Text))))>
      additionalInfoTitle
    </Styled(Connect(WithTheme(Styled(Text))))>
    <Styled(Connect(WithTheme(Styled(Text)))) />
  </Connect(WithTheme(Styled(View)))>
</Accordion>
`;
